src/main/java/com/google/api/codegen/util/testing/TestValueGenerator.java

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util.testing;
+
+import com.google.api.codegen.util.Name;
+import com.google.api.tools.framework.model.TypeRef;
+
+import java.util.HashMap;
+
+/**
+ * A utility class used by the test generators which populates values for primitive fields.
+ *
+ * <p>The populated values will be unique (except for boolean values) and deterministic based on the
+ * populated fields and the sequence.
+ */
+public abstract class TestValueGenerator {
+  protected HashMap<Name, String> valueTable = new HashMap<>();
+
+  public String getAndStoreValue(TypeRef type, Name identifier) {
+    if (!valueTable.containsKey(identifier)) {
+      String value = generateValue(type, identifier);
+      while (type.getPrimitiveTypeName() != "bool" && valueTable.containsValue(value)) {
+        // If the value already exists regenerate using a deterministically different identifier.
+        identifier = identifier.join("1");
+        value = generateValue(type, identifier);
+      }
+      valueTable.put(identifier, generateValue(type, identifier));
+    }
+    return valueTable.get(identifier);
+  }
+
+  protected abstract String generateValue(TypeRef typeRef, Name identifier);
```

Comment:
Fair enough. I agree that this is polymorphic behavior but it is a very simple one. I was thinking of using fewer classes since this ValueGenerator is a simple utility class so I was not very concerned about the decoupling and future changes. As you said, if the composition makes the code easier to understand and the number of classes is not an important matrix then composition maybe a better approach here.